effectsize_exemplar_within.Rmd

```diff
@@ -0,0 +1,137 @@
+data <- 
+  data %>% 
+  mutate(
+  response_time = 
+    id + # additive individual differences
+    layout * .4 + # main effect of layout
+    size * .2 + # main effect of size
+    color * 0 + 
+    layout * size * .5 + # 2-way interaction
+    size * color * 0 + 
+    layout * color * 0 + 
+    layout * size * color * 0 + 
+    rnorm(n()) # noise
+)
+```
+
+Even though we used numbers to simulate the model, the independent variables and subject ID are all factors.
+```{r within-factor}
+data <- 
+  data %>% 
+  mutate(
+    id = factor(id), 
+    layout = factor(layout), 
+    size = factor(size), 
+    color = factor(color)
+  )
+```
+
+
+### Compute effect sizes
+While **Cohen's *d* ** is often used for simple 2-factor, single-trial, between-subject designs, more complex designs can be more consistently interpretted with the **eta squared ($\eta^{2}$)** family of effect sizes, which represent the proportion of variance accounted for by a particular variable. A variant, **generalized eta squared ($\eta_{G}^{2}$)**, is particularly suited for providing comparable effect sizes in both between and within-subject designs [@Olejnik2003; @Bakeman2005]. This property makes it more easily applicable to meta-analyses.
+
+For those accustomed to Cohen's *d*, it's important to be aware that $\eta_{G}^{2}$ is typically smaller, with a Cohen's d of 0.2 being equivalent to a $\eta_{G}^{2}$ of around 0.02. Also, the actual number has little meaning beyond its scale relative to other effects. 
+
+```{r within-anova}
+results = afex::aov_ez(
+  data = data, 
+  id = 'id', # subject id column
+  dv = 'response_time', # dependent variable
+  within = c('layout', 'size', 'color'), # within-subject independent variables
+  between = NULL ,# between-subject independent variables
+  anova_table = list(es = 'ges') # effect size = generalized eta squared
+)
+```
+
+*Note: the warning indicates that the aov_ez function automatically collapses repetitions into a mean, which may be a problem if an experiment is not fully counterbalanced. This example, however, has every subject running in every combination of conditions, so simple collapsing is the correct procedure.*
+
+```{r  within-anova-cleanup}
+anovaResults <- 
+  results$anova_table %>% 
+  rownames_to_column('effect') %>%  # put effect names in a column
+  select(-`Pr(>F)`) # no need to show p-values
+  
+anovaResults %>% 
+  tidy() %>% 
+  rename(F = statistic)
+```
+
+*Note that the fractional degrees of freedom result from a Greenhousse-Geisser sphericity correction.*
+
+```
+TODO: Boostrapped 95%CIs for effect sizes
+Pro: people should
+Con: would make the guide even longer
+Maybe push into another guideline?
+```
+
+### Reporting the results
+
+Looking at the `F` and `ges` (generalized eta squared) columns, there are clear main effects for `layout` and `size` and an interaction between `layout` and `size`. However `color` and the other 2-way and 3-way interactions show only negligeable effects.
```

Comment:
@steveharoz I'm not sure I understand how the practical magnitude of the effects are being assessed here. Or at least, the current text seems to run close to being counter to our FAQ recommendations of not using pre-defined thresholds for effect sizes.

Are there previous studies in this domain that have found effects of a certain size to be meaningful? Some kind of benchmark against which to judge them? 

I guess I'm not entirely sure a standardized effect size is what I'd want to see reported in this context. Since this is a study of response times, I'd rather see estimated differences in response times between conditions. A measure of the proportion of variance in response times accounted for *some* layout or size (unknown without contrasts) I find more difficult to reason about.